### PR TITLE
Check for null on imex method and pass null to core if needed

### PR DESCRIPTION
**Pull request by Jikstra**
This maybe needs refactoring, i would like to split out parts into the `napi-macros-extensions.h`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -383,6 +383,7 @@ class DeltaChat extends EventEmitter {
   }
 
   importExport (what, param1, param2) {
+    if (!param2) param2 = null
     binding.dcn_imex(this.dcn_context, what, param1, param2)
```
Comment (ralphtheninja) [inline index.js]:
```suggestion
    binding.dcn_imex(this.dcn_context, what, param1 || '', param2 || '')
```

File: src/module.c [hunk 2]
```diff
@@ -885,7 +885,20 @@ NAPI_METHOD(dcn_imex) {
...
+  NAPI_STATUS_THROWS(napi_typeof(env, argv[3], &valuetype_argv3));
+
+  if(valuetype_argv3 != napi_null) {
+    size_t param2_size = 0;
+    NAPI_STATUS_THROWS(napi_get_value_string_utf8(env, argv[3], NULL, 0, &param2_size))
+    param2 = (char*)malloc((param2_size + 1) * sizeof(char));
+    size_t param2_len;
+    NAPI_STATUS_THROWS(napi_get_value_string_utf8(env, argv[3], param2, param2_size + 1, &param2_len))
+    param2[param2_size] = '\0';
+  }
```
Comment (ralphtheninja) [inline src/module.c]: This is unnecessarily complicated. See https://github.com/deltachat/deltachat-node/blob/master/src/module.c#L703 as an example

File: index.js [hunk 3]
```diff
@@ -383,6 +383,7 @@ class DeltaChat extends EventEmitter {
   }
 
   importExport (what, param1, param2) {
+    if (!param2) param2 = null
```
Comment (ralphtheninja) [inline index.js]:
```suggestion
    
```

**Comment by ralphtheninja** [comment]
Pass empty string if falsey and on c-side check for empty string and pass `NULL`.

File: src/module.c (hunk 2 above)
Comment (karissa) [reply src/module.c]: looks better now!

**Comment by karissa** [comment]
lgtm

**Comment by ralphtheninja** [comment]
Nice!

**Comment by ralphtheninja** [comment]
0.28.2

**Pull request merged**